geode-core/src/main/java/org/apache/geode/redis/internal/executor/hash/HSetExecutor.java

```diff
@@ -55,9 +69,11 @@ public void executeCommand(Command command, ExecutionHandlerContext context) {
     Object oldValue;
 
     if (onlySetOnAbsent())
-      oldValue = keyRegion.putIfAbsent(field, new ByteArrayWrapper(value));
+      oldValue = map.putIfAbsent(field, new ByteArrayWrapper(value));
```

Comment:
This will fail with `ConcurrentModificationException` if other connections try to access the same hash at the same time. I think we need to clone and then put, which will give us eventual consistency but not ACID.